contracts/test/v0.7/KeeperRegistryDev.test.ts

```diff
@@ -1991,27 +1984,94 @@ describe('KeeperRegistry', () => {
     })
   })
 
+  describe('#setPeerRegistryMigrationPermission() / #getPeerRegistryMigrationPermission()', () => {
+    const peer = randomAddress()
+    it('allows the owner to set the peer registries', async () => {
+      let permission = await registry.getPeerRegistryMigrationPermission(peer)
+      expect(permission).to.equal(0)
+      await registry.setPeerRegistryMigrationPermission(peer, 1)
+      permission = await registry.getPeerRegistryMigrationPermission(peer)
+      expect(permission).to.equal(1)
+      await registry.setPeerRegistryMigrationPermission(peer, 2)
+      permission = await registry.getPeerRegistryMigrationPermission(peer)
+      expect(permission).to.equal(2)
+    })
+    it('reverts if not called by the owner', async () => {
+      await expect(
+        registry.connect(admin).setPeerRegistryMigrationPermission(peer, 1),
+      ).to.be.revertedWith('Only callable by owner')
+    })
+  })
+
+  describe('migrateUpkeeps() / #receiveUpkeeps()', async () => {
+    context('when permissions are set', () => {
+      beforeEach(async () => {
+        await linkToken.connect(owner).approve(registry.address, toWei('100'))
+        await registry.connect(owner).addFunds(id, toWei('100'))
+        await registry.setPeerRegistryMigrationPermission(registry2.address, 1)
+        await registry2.setPeerRegistryMigrationPermission(registry.address, 2)
+      })
+
+      it('migrates an upkeep', async () => {
+        expect((await registry.getUpkeep(id)).balance).to.equal(toWei('100'))
+        expect((await registry.getUpkeep(id)).checkData).to.equal(randomBytes)
+        expect(await registry.getUpkeepCount()).to.equal(1)
+        // migrate
+        await registry.connect(admin).migrateUpkeeps([id], registry2.address)
```

Comment:
Can you add a check for non admin failing to migrate.
Also owner succeeding to migrate